src/Microsoft.VisualStudio.Services.Agent/VstsAgentWebProxy.cs

```diff
@@ -141,8 +141,15 @@ public void LoadProxyBypassList()
 
             foreach (string bypass in proxyBypassEnv.Split(new [] {',', ';'}).Where(value => !string.IsNullOrWhiteSpace(value)).Select(value => value.Trim()))
             {
-                Trace.Info($"Bypass proxy for: {bypass}.");
-                ProxyBypassList.Add(bypass);
+                string saveRegexString = bypass;
+
+                var regExp = new Regex("(?<!\\\\)([.])(?!\\*)");
```

Comment:
I believe we should cover this new processing logic by unit tests.

Reply:
Added test for new logic